ansible_ai_connect/ai/api/model_pipelines/registry.py

```diff
@@ -44,14 +46,11 @@ def __init__(self, api_type: t_model_mesh_api_type):
         self.api_type = api_type
 
     def __call__(self, cls):
-        if issubclass(cls, ModelPipelineCompletions):
-            PIPELINES[self.api_type][ModelPipelineCompletions] = cls
-        elif issubclass(cls, ModelPipelineContentMatch):
-            PIPELINES[self.api_type][ModelPipelineContentMatch] = cls
-        elif issubclass(cls, ModelPipelinePlaybookGeneration):
-            PIPELINES[self.api_type][ModelPipelinePlaybookGeneration] = cls
-        elif issubclass(cls, ModelPipelinePlaybookExplanation):
-            PIPELINES[self.api_type][ModelPipelinePlaybookExplanation] = cls
-        elif issubclass(cls, MetaData):
+        for pipe in EMPTY_PIPE.keys():
```

Comment:
I'm not sure this is any easier to read.. 

If you consider it better then I don't mind.

Reply:
Agree, it was easier to read, the benefit of new implementation is that you shouldn't know/look for the registration code. You just add new pipeline in one place and it works.

Reply:
Thank-you @hasys for the explanation.

I agree, your new implementation does make it more extensible :+1: